soroban-env-host/src/host/frame.rs

```diff
@@ -525,9 +525,13 @@ impl Host {
 
         self.fn_call_diagnostics(id, &func, args)?;
 
+        // Try dispatching the contract to the compiled-in registred
+        // implmentation. Only the contracts with the special (empty) executable
+        // are dispatched in this way, so that it's possible to switch the
+        // compiled-in implementation back to Wasm via
+        // `update_current_contract_wasm`.
         // "testutils" is not covered by budget metering.
-        #[cfg(any(test, feature = "testutils"))]
-        {
+        if cfg!(any(test, feature = "testutils")) && self.is_test_contract_executable(id)? {
```

Comment:
❗ I believe this change, using `cfg!` with an `if` block instead of the attribute on the block, will result in the block being compiled for both test and non-test builds. There might not be a consequence of that, but given that it could be a reasonable assumption that no test code ever end up in a production build, I think it's worth keeping it an attribute on the block so it is never compiled on prod builds.

Reply:
reverted